### PR TITLE
Improve PyTorch profiler flop computation formulas

### PR DESCRIPTION
**Pull request by xuzhao9**
Summary:

Improve the flops computation formula of aten::conv2d operator to support stride, pad, dilation, and groups arguments.

This diff also fixes the following issues:
- Apply a factor of 2 to aten::mm because output accounts for multiplication and addition.
- Fix incorrect names of scalar operators to aten::mul and aten::add.

Test Plan:

```python
python test/test_profiler.py
```

### CODE REVIEW

File: torch/csrc/autograd/profiler_utils.cpp [hunk 1]
```diff
@@ -115,23 +115,51 @@ std::unordered_map<std::string, c10::IValue> saveExtraArgs(const at::RecordFunct
...
+    const std::vector<int64_t> dilation = dilation_ref.toIntVector();
     if (input_sizes.size() != 4 || kernel_sizes.size() != 4) {
       TORCH_WARN("Failed to compute flops for op aten::conv2d because both input and weight must be size 4.");
       return 0;
     }
+    if (!groups) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because group size must not be 0.");
+      return 0;
+    }
+    if (padding.size() != 2 || dilation.size() != 2) {
```
Comment (jspark1105) [inline torch/csrc/autograd/profiler_utils.cpp:155]: PyTorch doesn't support having a different padding for left and right (or top and bottom)? Sorry I'm more familiar with Caffe2 conv operator.

**Review by jspark1105 — APPROVED** [review]
Thanks for incorporating changes! Accepting but it would be great if @ilia-cher can also take a look.

File: torch/csrc/autograd/profiler_utils.cpp (hunk 1 above)
Comment (xuzhao9) [reply torch/csrc/autograd/profiler_utils.cpp:155]: According to torch.nn.conv2d doc, padding only supports number of points in two directions and is applied on both sides: https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html

**Review by ilia-cher — APPROVED** [review]
